pkg/destroy/azure/azure.go

```diff
@@ -100,26 +101,85 @@ func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.
 }
 
 // Run is the entrypoint to start the uninstall process.
-func (o *ClusterUninstaller) Run() error {
+func (o *ClusterUninstaller) Run() (err error) {
+	var errs []error
+
 	o.configureClients()
 
-	o.Logger.Debug("deleting public records")
-	if err := deletePublicRecords(context.TODO(), o.zonesClient, o.recordsClient, o.privateZonesClient, o.privateRecordSetsClient, o.Logger, o.ResourceGroupName); err != nil {
+	group := o.InfraID + "-rg"
+	timeout := 30 * time.Minute
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	wait.Until(
+		func() {
+			o.Logger.Debugf("deleting public records")
+			err = deletePublicRecords(ctx, o.zonesClient, o.recordsClient, o.privateZonesClient, o.privateRecordSetsClient, o.Logger, group)
+			if err != nil {
+				o.Logger.Debug(err)
+				return
+			}
+			cancel()
+		},
+		timeout,
```

Comment:
secondly this argument is period after which the function is called and not the timeout..

> https://godoc.org/k8s.io/apimachinery/pkg/util/wait#Until
```
Until loops until stop channel is closed, running f every period.

Until is syntactic sugar on top of JitterUntil with zero jitter factor and with sliding = true (which means the timer for period starts after the f completes).
```

Reply:
Fixed.